smack-websocket/src/main/java/org/jivesoftware/smack/websocket/WebsocketConnectionAttemptState.java

```diff
@@ -0,0 +1,94 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jivesoftware.smack.websocket;
+
+import java.lang.reflect.InvocationTargetException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import org.jivesoftware.smack.SmackException.EndpointConnectionException;
+import org.jivesoftware.smack.c2s.internal.ModularXmppClientToServerConnectionInternal;
+import org.jivesoftware.smack.fsm.StateTransitionResult;
+import org.jivesoftware.smack.websocket.XmppWebsocketTransportModule.EstablishingWebsocketConnectionState;
+import org.jivesoftware.smack.websocket.implementations.WebsocketImplProvider;
+import org.jivesoftware.smack.websocket.implementations.WebsocketInterface;
+import org.jivesoftware.smack.websocket.implementations.okhttp.OkHttpWebsocketImpl;
+import org.jivesoftware.smack.websocket.rce.WebsocketRemoteConnectionEndpoint;
+
+public final class WebsocketConnectionAttemptState {
+
+    private final ModularXmppClientToServerConnectionInternal connectionInternal;
+    private final XmppWebsocketTransportModule.XmppWebsocketTransport.DiscoveredWebsocketEndpoints discoveredEndpoints;
+
+    EndpointConnectionException connectionException;
+
+    private WebsocketRemoteConnectionEndpoint connectedEndpoint;
+
+    WebsocketInterface websocket;
+
+    WebsocketConnectionAttemptState(ModularXmppClientToServerConnectionInternal connectionInternal,
+                    XmppWebsocketTransportModule.XmppWebsocketTransport.DiscoveredWebsocketEndpoints discoveredWebsocketEndpoints,
+                    EstablishingWebsocketConnectionState establishingWebsocketConnectionState) {
+        assert discoveredWebsocketEndpoints != null;
+        this.connectionInternal = connectionInternal;
+        this.discoveredEndpoints = discoveredWebsocketEndpoints;
+    }
+
+    StateTransitionResult.Failure establishWebsocketConnection() throws InterruptedException {
+        List<WebsocketRemoteConnectionEndpoint> endpoints = discoveredEndpoints.result.discoveredRemoteConnectionEndpoints;
+        List<Throwable> connectionFailureList = new ArrayList<>();
+
+        try {
+            // Obtain desired websocket implementation by using WebsocketImplProvider
+            websocket = WebsocketImplProvider.getWebsocketImpl(OkHttpWebsocketImpl.class, connectionInternal, discoveredEndpoints);
+        } catch (NoSuchMethodException | SecurityException | InstantiationException |
+                IllegalAccessException | IllegalArgumentException | InvocationTargetException exception) {
+            Logger.getAnonymousLogger().log(Level.WARNING, "Couldn't obtain the underlying websocket implementation: ", exception);
```

Comment:
Do not use an anonymous logger. In fact, I think this log statement should be removed, as the state transition results should already be subject to logging.